### ISSUE TITLE
Replace `cudaStreamDefault`

### ISSUE THREAD

**Issue by gaohao95** [rapidsai/distributed-join#46]
`cudaStreamDefault` is a flag passed to `cudaStreamCreateWithFlags` instead of a valid stream. We should replaced that with either `rmm::cuda_stream_default` or `0`.

**Comment by nsakharnykh** [comment]
I can't find `cudaStreamCreateWithFlags` in our code - did you mean that `cudaStreamDefault` is passed into RMM functions (allocate/deallocate), and we should replace it with `rmm::cuda_stream_default`?

**Comment by gaohao95** [comment]
> I can't find `cudaStreamCreateWithFlags` in our code - did you mean that `cudaStreamDefault` is passed into RMM functions (allocate/deallocate), and we should replace it with `rmm::cuda_stream_default`?

Yes. Basically we shouldn't pass `cudaStreamDefault` to a function that accepts `cudaStream_t` or `rmm::cuda_stream`.
